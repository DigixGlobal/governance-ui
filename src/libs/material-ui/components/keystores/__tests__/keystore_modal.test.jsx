import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import KeystoreModal from '../keystore_modal';

jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

const keystoreTypes = [{
  id: 1,
  name: 'keystore test 1',
  icon: 'testIcon',
  color: 'blue',
  subtitle: 'test subtitle',
}, {
  id: 2,
  name: 'keystore test 2',
  icon: 'testIcon2',
  color: 'black',
  subtitle: 'test subtitle 2',
}];

describe('<KeystoreModal />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <KeystoreModal
        submitFunc={jest.fn}
        keystoreTypes={keystoreTypes}
        form={() => <div />}
        trigger={<div />}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
