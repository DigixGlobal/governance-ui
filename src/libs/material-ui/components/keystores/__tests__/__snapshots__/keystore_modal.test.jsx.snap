// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeystoreModal /> renders correctly 1`] = `
<EZModal
  content={[Function]}
  data={
    Object {
      "networks": undefined,
      "tokens": Array [],
    }
  }
  error={false}
  handleSubmit={[Function]}
  header={[Function]}
  loading={false}
  noSubmitButton={[Function]}
  onClose={[Function]}
  onReset={[Function]}
  trigger={<div />}
/>
`;
